### ISSUE TITLE
The odata v2 metadata info missed extend elements in our extensibility testing

### ISSUE THREAD

**Issue by shaozhm** [cap-js-community/odata-v2-adapter#14]
Hi Colleagues,

Recently Our Project (https://github.wdf.sap.corp/ctsm/ctsm) has been doing extensibility testing according to the CAP document (https://pages.github.tools.sap/cap/docs/guides/extensibility/). And we found an issue with the metadata info missing in the converted ODATA v2. (It missed all-new extended fields and extended Fiori Element annotations).

### Symptom
1. You can see our `extensions.cds` file (https://github.wdf.sap.corp/ctsm/ctsm/pull/6845/files#diff-5d4631d6f9245af3077ccd10ec33112611ab884ef208e553b2b5f4a8b00f845a)
   - We added a new field `Z_ext_field` into an existing entity `Studies`.
   - We added an annotation for Fiori Element UI (FieldGroup #ExtendData)
  
2. Then extend a specific tenant t1. (local testing steps: just use `cds subscribe`, `cds pull`, `cds build` and `cds push`)
3. We check the metadata of OData V4 - there are all extended elements. **But, OData V2, which was converted by the adapter lost the extended elements**.

![image](https://github.com/cap-js-community/odata-v2-adapter/assets/13991457/7f940068-11e4-42ee-8587-a1fb473f126c)

Could you have a check? I think it is an issue. Thanks

BR,
Zhimin Shao

**Comment by shaozhm** [comment]
Thanks @oklemenz2 ,
According to the code you mentioned, We debugged our project and resolved the problem. 
We found `"cds.xt.ModelProviderService": true,` must be declared explicitly whether `"extensibility": true,` is declared.

code adjustment refer to:
https://github.wdf.sap.corp/ctsm/ctsm/blob/feature/evaluation_extensibility/package.json#L229-L230

But according to cap document (https://pages.github.tools.sap/cap/docs/guides/multitenancy/mtxs#model-provider-config), if we declare `"extensibility": true,`, `ModelProviderService` is activated and configured automatically.

So maybe it could be an improvement point, just a suggestion from my side.

Thank you very much again.

BR,
Zhimin Shao
